sanic/testing.py

```diff
@@ -391,33 +215,39 @@ def __init__(
         response.body = response.content
         response.content_type = response.headers.get("content-type")
 
-        if hasattr(response, "return_value"):
-            request = response.return_value
-            del response.return_value
-            return request, response
+        return self.last_request, response
+
+    async def websocket(self, uri, subprotocols=None, *args, **kwargs):
+        scheme = "ws"
+        path = uri
+        root_path = "{}://{}".format(scheme, ASGI_HOST)
+
+        headers = kwargs.get("headers", {})
+        headers.setdefault("connection", "upgrade")
+        headers.setdefault("sec-websocket-key", "testserver==")
+        headers.setdefault("sec-websocket-version", "13")
+        if subprotocols is not None:
+            headers.setdefault(
+                "sec-websocket-protocol", ", ".join(subprotocols)
+            )
+
+        scope = {
+            "type": "websocket",
+            "asgi": {"version": "3.0"},
```

Comment:
should we expose the version or just hardcoded ?

Reply:
It really is not important for this purpose. This scope is merely a placeholder to allow the handshake and connection to be established. I think we are safe to leave it hardcoded.